src/fancy-tools/rotateTool.js

```diff
@@ -0,0 +1,90 @@
+/* eslint no-loop-func: 0 */ // --> OFF
+import external from './../externalModules.js';
+import baseTool from './../base/baseTool.js';
+import angleBetweenPoints from './shared/angleBetweenPoints.js';
+
+const defaultStrategy = (evt) => {
+  const eventData = evt.detail;
+  const { element, viewport } = eventData;
+  const initialRotation = viewport.initialRotation;
+
+  // Calculate the center of the image
+  const rect = element.getBoundingClientRect(element);
+  const { clientWidth: width, clientHeight: height } = element;
+
+  const initialPoints = {
+    x: eventData.startPoints.client.x,
+    y: eventData.startPoints.client.y
+  };
+  const { scale, translation } = viewport;
+  const centerPoints = {
+    x: rect.left + width / 2 + translation.x * scale,
+    y: rect.top + height / 2 + translation.y * scale
+  };
+
+  const currentPoints = {
+    x: eventData.currentPoints.client.x,
+    y: eventData.currentPoints.client.y
+  };
+
+  const angleInfo = angleBetweenPoints(centerPoints, initialPoints, currentPoints);
+
+  if (angleInfo.direction < 0) {
+    angleInfo.angle = -angleInfo.angle;
+  }
+
+  viewport.rotation = initialRotation + angleInfo.angle;
+};
+
+const horizontalStrategy = (evt) => {
+  const eventData = evt.detail;
+  const { viewport, deltaPoints } = eventData;
```

Comment:
Man, I love destructuring so much.

Reply:
Started to like it a lot as well. hahaha